app/src/main/java/com/gophillygo/app/FilterDialog.java

```diff
@@ -62,58 +67,33 @@ public void onDismiss(DialogInterface dialog) {
-
-        Filter filter = null;
+        filter = null;
         if (getArguments() != null) {
-            filter = getArguments().getParcelable(FILTER_ARG);
+            filter = (Filter) getArguments().getSerializable(FILTER_ARG);
         }
         if (filter == null) {
             filter = new Filter();
         }
 
-        // set up filter buttons
-        natureButton = dialog.findViewById(R.id.filter_modal_nature_category_button);
-        natureButton.setChecked(filter.nature);
-        exerciseButton = dialog.findViewById(R.id.filter_modal_excercise_category_button);
-        exerciseButton.setChecked(filter.exercise);
-        educationalButton = dialog.findViewById(R.id.filter_modal_educational_category_button);
-        educationalButton.setChecked(filter.educational);
-        beenButton = dialog.findViewById(R.id.filter_modal_been_button);
-        beenButton.setChecked(filter.been);
-        wantToGoButton = dialog.findViewById(R.id.filter_modal_want_to_go_button);
-        wantToGoButton.setChecked(filter.wantToGo);
-        notInterestedButton = dialog.findViewById(R.id.filter_modal_not_interested_button);
-        notInterestedButton.setChecked(filter.notInterested);
-        likedButton = dialog.findViewById(R.id.filter_modal_liked_button);
-        likedButton.setChecked(filter.liked);
-        accessibleButton = dialog.findViewById(R.id.filter_modal_accessible_button);
-        accessibleButton.setChecked(filter.accessible);
+        Dialog dialog = super.onCreateDialog(savedInstanceState);
+        FilterModalBinding binding = DataBindingUtil.inflate(LayoutInflater.from(getContext()),
+                R.layout.filter_modal, null, false);
+        binding.setFilter(filter);
+        binding.setDialog(this);
+        dialog.setContentView(binding.getRoot());
 
+        // set up filter buttons
         filterButtons = new ArrayList<>(8);
-        filterButtons.add(natureButton);
-        filterButtons.add(exerciseButton);
-        filterButtons.add(educationalButton);
-        filterButtons.add(beenButton);
-        filterButtons.add(wantToGoButton);
-        filterButtons.add(notInterestedButton);
-        filterButtons.add(likedButton);
-        filterButtons.add(accessibleButton);
-
-        // set up modal function buttons
-        doneButton = dialog.findViewById(R.id.filter_modal_done_button);
-        resetButton = dialog.findViewById(R.id.filter_modal_reset_button);
-
-        doneButton.setOnClickListener(v -> dismiss());
-
-        resetButton.setOnClickListener(v -> {
-            for (GpgToggleButton button: filterButtons) {
-                button.setChecked(false);
-            }
-        });
+        for (int id : FILTER_BUTTONS) {
+            filterButtons.add(dialog.findViewById(id));
+        }
 
         return dialog;
     }
+
+    public void resetFilters(View view) {
+        for (GpgToggleButton button : filterButtons) {
+            button.setChecked(false);
```

Comment:
How about clearing the `Filter` model properties here, instead of the button state? The `checked` property of the button should update then from the model (instead of the other way around). Then it won't be necessary to look up or track the views here, since we can rely on data binding. A `reset` method on the `Filter` model should do the trick.